pkg/types/automation/basetypes.go

```diff
@@ -0,0 +1,349 @@
+package automation
+
+import (
+	"encoding/hex"
+	"encoding/json"
+	"fmt"
+	"math/big"
+	"strings"
+	"time"
+)
+
+const (
+	checkResultDelimiter = 0x09
+)
+
+// checkResultStringTemplate is a JSON template, used for debugging purposes only.
+var checkResultStringTemplate = `{
+	"PipelineExecutionState":%d,
+	"Retryable":%v,
+	"Eligible":%v,
+	"IneligibilityReason":%d,
+	"UpkeepID":%s,
+	"Trigger":%s,
+	"WorkID":"%s",
+	"GasAllocated":%d,
+	"PerformData":"%s",
+	"FastGasWei":%s,
+	"LinkNative":%s,
+	"RetryInterval":%d
+}`
+
+func init() {
+	checkResultStringTemplate = strings.Replace(checkResultStringTemplate, " ", "", -1)
+	checkResultStringTemplate = strings.Replace(checkResultStringTemplate, "\t", "", -1)
+	checkResultStringTemplate = strings.Replace(checkResultStringTemplate, "\n", "", -1)
```

Comment:
`ReplaceAll` is available:
```suggestion
	checkResultStringTemplate = strings.ReplaceAll(checkResultStringTemplate, " ", "")
	checkResultStringTemplate = strings.ReplaceAll(checkResultStringTemplate, "\t", "")
	checkResultStringTemplate = strings.ReplaceAll(checkResultStringTemplate, "\n", "")
```
Although this seems odd. Is the point to keep it in human readable form in source?

Reply:
Oh and there is a `Replacer`:
```suggestion
	checkResultStringTemplate = strings.NewReplacer(" ", "", "\t", "", "").Replace(checkResultStringTemplate)
```

Reply:
Yeah so the init implementation is just to collapse the more readable source string into a single line; my understanding here is that using a JSON encoder was deemed to be too expensive at the time, and using a string template was cheaper/faster. I didn't want to modify any of the code lifted out of automation in this PR; I wanted to keep everything as it was written in the automation repo to keep the impacts of these changes to a minimum